recipe/build.sh

```diff
@@ -11,6 +11,12 @@ if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1" ]]; then
     CMAKE_ARGS="$CMAKE_ARGS -DLLVM_CONFIG_PATH=$BUILD_PREFIX/bin/llvm-config -DMLIR_TABLEGEN_EXE=$BUILD_PREFIX/bin/mlir-tblgen"
 fi
 
+if [[ "$variant" == "emscripten"* ]]; then
+    CMAKE_ARGS="$CMAKE_ARGS \
+        -DLLVM_DEFAULT_TARGET_TRIPLE=wasm32-unknown-emscripten \
+        -DLLVM_TARGETS_TO_BUILD=WebAssembly"
+fi
```

Comment:
Why create a variant to target WebAssembly? I'd rather you enable WebAssembly for all builds of flang.

Reply:
I was mostly trying to leave the other builds intact. But if it's okay to enable WebAssembly for all the builds, then I'm okay with that too. It would make it easier also :)

Reply:
So this should move out of the if branch (and into the general CMake invocation below), and we need to expand the targets to at least:
```
-DLLVM_TARGETS_TO_BUILD="X86;WebAssembly" \
```
If we're not building variants (fine by me), then I don't think we should set `-DLLVM_DEFAULT_TARGET_TRIPLE`, but rather the emscripten side will have to provide `-triple` or `-target` in the compiler activation flags.

Reply:
Yeah; especially because we're not actually rebuilding llvm (the library) here - but WebAssembly is being built as [part of](https://github.com/conda-forge/llvmdev-feedstock/blob/d552ed563ba6f335cf9bcd4be3ae73d8b0b2ce3d/recipe/build.sh#L57) our `llvmdev` builds already, so that should be fine.